### PR TITLE
fix(providers): adaptation to breaking change in google api

### PR DESCRIPTION
**Pull request by flyingtof**
Refactored googleMap's Marker.js to follow google maps API changes (use
google.maps.Marker#setLabel)
Removed googleMap's Marker.js as it's not relevant anymore
Removed irrelevant listener in googleMap.js
added end to end test for google map
bumped versions and configs in order to perform end to end tests using
jest and puppeter
updated documentation

Closes #19237

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -173,7 +173,8 @@ Here are the common options:
             anchor: new google.maps.Point(0, 32)
         },
         label: function(point) {
-            return '<div class="map-custom-label">' + point.id + '</div>';
+            // a function that return the text value of the label
+            return `${point.data.location.name}`;
```
Comment (ybonnefond) [inline README.md]: Why using string literals here?
Either just return `point.data.location.name` or `String(point.data.location.name)` if you want to cast it to string (be careful of casting null, arrays, objects, functions, etc...)

File: src/providers/googleMap/Marker.js [hunk 2]
```diff
@@ -1,52 +1,62 @@
...
         }
+    }
 
-        if (marker.draggable) {
-            google.maps.event.addListener(this, 'drag', (event) => {
-                if (options.label) {
-                    this.label.update(event.latLng);
-                }
+    addOnDragEndEventListener(marker, options) {
+        if (marker.draggable && options.onDragEnd) {
```
Comment (ybonnefond) [inline src/providers/googleMap/Marker.js]: Avoid truthy checks

File: src/providers/googleMap/Marker.js [hunk 3]
```diff
@@ -1,52 +1,62 @@
...
+    }
 
-        if (options.label) {
-            this.label = new Label({
-                map: map,
-                position: this.getPosition()
-            }, point, options.label);
+    defineLabel(options, point) {
+        let label = options.label;
+        if (label) {
```
Comment (ybonnefond) [inline src/providers/googleMap/Marker.js]: Avoid truthy checks

File: src/providers/googleMap/Marker.js [hunk 4]
```diff
@@ -1,52 +1,62 @@
...
+}
+
+function buildMarkerOptions(point, map, options) {
+    let marker = {
+        position: new google.maps.LatLng(point.latitude, point.longitude),
+        map: map,
+        draggable: options.draggable || false
+    };
+
+    if (options.icon) {
```
Comment (ybonnefond) [inline src/providers/googleMap/Marker.js]: Avoid truthy checks

File: test/commons/server.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+class Server {
+    constructor() {
+        const app = express();
+        app.use('/', express.static(__dirname + '/../../dist/'));
+        this.app = app;
+    }
+
+    start(staticContent, config) {
+        if (staticContent) {
```
Comment (ybonnefond) [inline test/commons/server.js]: Avoid truthy checks

File: src/utils/type.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+function isAString(something) {
+    return typeof something === 'string';
+}
+
+function isAFunction(something) {
+    return typeof something === 'function';
+}
+
+function exists(something) {
```
Comment (ybonnefond) [inline src/utils/type.js]: `isDefined`?

File: README.md (hunk 1 above)
Comment (flyingtof) [reply README.md]: You're right. Just return point.data.location.name

File: src/providers/googleMap/Marker.js (hunk 2 above)
Comment (flyingtof) [reply src/providers/googleMap/Marker.js]: Replace 
if (something) { ...
by 
if (defined(something)) { ...
using function defined from simple-js-validator

File: src/providers/googleMap/Marker.js (hunk 3 above)
Comment (flyingtof) [reply src/providers/googleMap/Marker.js]: see previous comment

File: src/providers/googleMap/Marker.js (hunk 4 above)
Comment (flyingtof) [reply src/providers/googleMap/Marker.js]: see previous comment

File: test/commons/server.js (hunk 5 above)
Comment (flyingtof) [reply test/commons/server.js]: see previous comment

File: src/utils/type.js (hunk 6 above)
Comment (flyingtof) [reply src/utils/type.js]: Yes. Remove this function and use function 'defined' from simple-js-validator

**Review by kavrillon — APPROVED** [review]

**Review by ybonnefond — APPROVED** [review]

**Comment by ybonnefond** [comment]
The point of my "thruthy check" comments is to avoid having the `if` evaluates whether the condition is truthy or not

```javascript
// Thruthy check
if (variable) { /* ... */ }
if (isDefined(variable)) { /* ... */ }
```
In this case you don't know what your if is testing unless you go into isDefined definition and check it returns a boolean. But the . condition can still be met if someone change isDefined and it starts returning 1, "false", "-1", [], {}, etc... 


```javascript
if (true === variable)  { /* ... */ }
if ('string' === typeof variable && variable.length > 0)  { /* ... */ }
if (true === isString(variable))  { /* ... */ }
```
In this case, condition is explicit, no ambiguity

https://dorey.github.io/JavaScript-Equality-Table/

**Pull request merged**